src/Engine/ProtoCore/Lang/BuiltInMethods.cs

```diff
@@ -211,6 +216,7 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("array2", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank))
                     },
                     ID = BuiltInMethods.MethodID.kConcat,
+                    MethodAttributes = new MethodAttributes(){Description  = Resources.ReturnsConcatenatingArray}
```

Comment:
This is same as List.Join but it fails in certain cases, whereas List.Join works better. Let's hide this method, by passing isHiddenInLibrary as true. 
Update the documentation to "Joins two lists into a single list."